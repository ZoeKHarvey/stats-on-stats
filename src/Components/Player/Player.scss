.player {
  font-family: 'Barlow Condensed', sans-serif;
  display: flex;
  justify-content: space-evenly;
  height: 87vh;
  

}

ul {
  list-style: none;
}

.player__h2--name {
  display: block;
}

.player__img--ice {
  position: absolute;
  z-index: -100;
  height: 100vh;
  width: 100%;
}

.card__player--active {
  font-size: 3em;
}