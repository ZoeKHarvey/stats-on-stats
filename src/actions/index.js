export const getTeams = (teams) => ({
  type:'GET_TEAMS',
  teams
})

export const getStandings = (standings) => ({
  type: 'GET_STANDINGS',
  standings
})

export const getRoster = (roster) => ({
  type: 'GET_ROSTER',
  roster
})

export const getPlayer = (player) => ({
  type: 'GET_PLAYER',
  player
})

export const getPlayerStats = (player) => ({
  type: 'GET_PLAYER_STATS',
  player
})

export const getTeamSchedule = (schedule) => ({
  type: 'GET_TEAM_SCHEDULE',
  schedule
})

export const getPlayerProjections = (player) => ({
  type: 'GET_PLAYER_PROJECTIONS',
  player
})

export const favoriteTeams = (newTeam) => ({
  type: 'GET_FAVORITE_TEAMS',
  newTeam
})

export const isLoading = (isLoading) => ({
  type: 'IS_LOADING',
  isLoading
})

export const hasError = (errorMsg) => ({
  type: 'HAS_ERROR',
  errorMsg
})



