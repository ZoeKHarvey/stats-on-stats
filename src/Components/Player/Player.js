import { getPlayerStats } from '../../actions';
import React, {Component} from 'react';
import { bindActionCreators } from 'redux';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux'
import { player } from '../../reducers/playerInfo';
import { fetchPlayerStats } from '../../apiCalls';

 export class Player extends Component {

  componentDidMount = async () => {
    const { playerInfo, getPlayerStats } = this.props;
    try {
      const stat = await fetchPlayerStats(playerInfo.id);
      getPlayerStats(this.organizeStats(stat))
      console.log('stats==>', stat)
    } catch (error) {
      console.log('error')
    }
  }

  organizeStats = (playerInfo) => {
    return {
      assists: playerInfo.assists,
      games: playerInfo.games,
      goals: playerInfo.goals,
      hits: playerInfo.hits,
      pim: playerInfo.pim,
      plusMinus: playerInfo.plusMinus,
      points: playerInfo.points,
      powerPlayPoints: playerInfo.powerPlayPoints,
      shotPct: playerInfo.shotPct,
      shots: playerInfo.shots,
    }
  }
 
  render() {
    const {playerInfo, playerStats} = this.props
    console.log('player stats in render', playerStats)
	return (
		<section className='player'>
      {playerInfo.fullName}
      {playerInfo.active}
      {playerInfo.birthCity}
      {playerInfo.birthCountry}
      {playerInfo.birthStateProvince}
      {playerInfo.captain}
      {playerInfo.currentAge}
      {playerInfo.height}
      {playerInfo.weight}
      {playerInfo.rosterStatus}
      {playerInfo.shootsCatches}
      {playerStats.assists}
      {playerStats.games}
      {playerStats.goals}
      {playerStats.hits}
      {playerStats.pim}
      {playerStats.plusMinus}
      {playerStats.points}
      {playerStats.powerPlayPoints}
      {playerStats.shotPct}
      {playerStats.shots}
		</section>

	)}
}

export const mapStateToProps = (state) => ({
  teams:state.teams,
  playerStats: state.playerStats,
  playerInfo:state.playerInfo
})

export const mapDispatchToProps = (dispatch) => (
  bindActionCreators({
    getPlayerStats
  }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(Player);