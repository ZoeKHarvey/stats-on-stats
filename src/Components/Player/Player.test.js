import React from 'react';
import { shallow } from 'enzyme'
import { Player, mapStateToProps, mapDispatchToProps } from './Player';
import { getPlayerStats, getPlayerProjections, favoritePlayer } from '../../actions/index';

describe('Player', () => {
  let wrapper;

  let playerInfo = {
    fullName: 'Vito Burrito',
  }

  let playerStats = {
    assists: 100,
    goals: 90
  }

  let playerProjections = {
    assists: 200
  }

  beforeEach(() => {
    wrapper = shallow(
      <Player 
      playerInfo={playerInfo}
      playerStats={playerStats}
      playerProjections={playerProjections} />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mockStateToProps', () => {
  it('should return an object with playerStats', () => {
    const mockState = {
      playerStats: {assists: 2},
      playerInfo: {fullName: 'dog'},
      playerProjections: {assists: 100}
    };
    const expected = {
      playerStats: {assists: 2},
      playerInfo: {fullName: 'dog'},
      playerProjections: {assists: 100}
    };

    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected)

  });

  const mockDispatch = jest.fn();

})