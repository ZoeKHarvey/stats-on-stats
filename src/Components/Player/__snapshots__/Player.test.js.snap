// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player should match the snapshot 1`] = `ShallowWrapper {}`;
